core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -421,6 +434,321 @@ private[spark] object ClosureCleaner extends Logging {
   }
 }
 
+private[spark] object IndylambdaScalaClosures extends Logging {
+  private val isScala2_11 = scala.util.Properties.versionString.contains("2.11")
+
+  // internal name of java.lang.invoke.LambdaMetafactory
+  val LambdaMetafactoryClassName = "java/lang/invoke/LambdaMetafactory"
+  // the method that Scala indylambda use for bootstrap method
+  val LambdaMetafactoryMethodName = "altMetafactory"
+  val LambdaMetafactoryMethodDesc = "(Ljava/lang/invoke/MethodHandles$Lookup;" +
+    "Ljava/lang/String;Ljava/lang/invoke/MethodType;[Ljava/lang/Object;)" +
+    "Ljava/lang/invoke/CallSite;"
+
+  /**
+   * Check if the given reference is a indylambda style Scala closure.
+   * If so (e.g. for Scala 2.12+ closures), return a non-empty serialization proxy
+   * (SerializedLambda) of the closure;
+   * otherwise (e.g. for Scala 2.11 closures) return None.
+   *
+   * @param maybeClosure the closure to check.
+   */
+  def getSerializationProxy(maybeClosure: AnyRef): Option[SerializedLambda] = {
+    if (isScala2_11) {
+      // Keep existing behavior in Spark 2.4: assume Scala 2.11 doesn't use indylambda.
+      // NOTE: It's actually possible to turn on indylambda in Scala 2.11 via delambdafy:inline,
+      //       but that's not the default and we don't expect it to be in use.
+      return None
+    }
```

Comment:
This is Spark 2.4-specific. It was there in the existing code in `ClosureCleaner.getSerializedLambda`, and I'm porting it over to the new code as well.